### ISSUE TITLE
chunked_vector_test fails to compile with recent boost versions

### ISSUE THREAD

**Issue by lkshminarayanan** [scylladb/scylladb#18746]
When compiled with boost version 1.82+, `chunked_vector_test` fails with the following error :

```
/Scylla/scylladb/test/boost/chunked_vector_test.cc:378:24: error: no member named 'optional' in namespace 'std'
    auto filler = std::optional<exception_safe_class>(checker);
                  ~~~~~^
/Scylla/scylladb/test/boost/chunked_vector_test.cc:378:33: error: 'exception_safe_class' does not refer to a value
    auto filler = std::optional<exception_safe_class>(checker);
                                ^
/Scylla/scylladb/test/boost/chunked_vector_test.cc:139:7: note: declared here
class exception_safe_class {
      ^
2 errors generated.
ninja: build stopped: subcommand failed.
```

This does compiles with older versions of boost. Although the code is unrelated to boost, the `optional` library seems to get implicitly included in older versions and not in the newer versions causing this issue. I have verified this by generating the dependency graph with [version 78](https://gist.github.com/lkshminarayanan/908be05bbca634526a60011dfbc03aab), which has the optional header in the list vs the graph with [version 83](https://gist.github.com/lkshminarayanan/151fe934e446fe93be33ca01074bcba4), which doesn't have that header in the list.

**Comment by lkshminarayanan** [comment]
Note that this is a followup to 4bbb66f, so it needs to be backported to older versions.

**Comment by lkshminarayanan** [comment]
Fixed by #18739
